internal/migrations/changeset_listener.go

```diff
@@ -0,0 +1,272 @@
+	for {
+		select {
+		case <-ctx.Done():
+			return nil
+		default:
+			if currentHeight >= cfg.EndHeight {
+				// Synced up with the old chain, no more changesets to pull. Close the listener
+				service.Logger.Info("changesets have been synchronized with the old chain", log.Int("height", currentHeight))
+				return nil
+			}
+
+			// Get the changeset metadata from the admin server
+			numChunks, size, err := adminClient.ChangesetMetadata(ctx, currentHeight)
+			if err != nil {
+				fmt.Println("Couldnt fetch changesets", err)
+				time.Sleep(retryDelay) // If no changeset is found, wait and try again
+				continue
+			}
+			btsReceived := int64(0)
+
+			service.Logger.Debug("received changeset metadata", log.Int("height", currentHeight), log.Int("numChunks", numChunks), log.Int("size", size))
+			wg := sync.WaitGroup{}
+			errChan := make(chan error, numChunks)
+
+			for i := int64(0); i < numChunks; i++ {
+				wg.Add(1)
+				go func(chunkIdx int64) {
+					for {
+						// Load the changeset from the admin server
+						cs, err := adminClient.LoadChangeset(ctx, currentHeight, chunkIdx)
+						if err != nil {
+							// If no changeset is found, wait and try again
+							time.Sleep(retryDelay)
+							continue
+						}
+
+						// Create the changeset migration event
+						evt := &ChangesetMigration{
+							Height:      big.NewInt(currentHeight),
+							TotalChunks: big.NewInt(numChunks),
+							ChunkIdx:    big.NewInt(chunkIdx),
+							Changeset:   cs,
+						}
+
+						csEvt, err := evt.MarshalBinary()
+						if err != nil {
+							service.Logger.Error("failed to marshal changeset migration event", log.String("error", err.Error()))
+							errChan <- err
+							return
+						}
+
+						service.Logger.Info("broadcasting changeset migration event", log.Int("height", currentHeight), log.Int("chunk", chunkIdx), log.Int("size", len(cs)))
+
+						// Broadcast the changeset migration event to the event store for voting
+						err = eventStore.Broadcast(ctx, ChangesetMigrationEventType, csEvt)
+						if err != nil {
+							errChan <- fmt.Errorf("failed to broadcast changeset migration event: %w", err)
+							return
+						}
+
+						btsReceived += int64(len(cs))
+						break
+					}
+					wg.Done()
+				}(i)
+			}
+
+			// Wait for all chunks to be successfully received
+			wg.Wait()
+			close(errChan)
+
+			// check errChan for all the errors
+			var errs error
+			for err := range errChan {
+				errs = errors.Join(errs, err)
+			}
+
+			if errs != nil {
+				return errs
+			}
+
+			if btsReceived != size {
+				return fmt.Errorf("incorrect changeset size received: %d, expected: %d", btsReceived, size)
+			}
+
+			if err = setLastStoredHeight(ctx, eventStore, currentHeight); err != nil {
+				return err
+			}
+			currentHeight++
+
+			// Delay before polling the admin server again for the next changeset
+			time.Sleep(retryDelay)
+		}
+	}
+}
+
+func (c *MigrationConfig) ExtractConfig(cfg map[string]string) error {
```

Comment:
I'm not 100% sure how we would achieve this, but it would be awesome to allow oracle configuration to be done as part of Kwild's "official" configuration, rather than relying on the double-nested map for user-defined extensions. 

I guess for now nbd since this is a different feature altogether, but it would be great if users could register extension configs as an "official flag" that would show up with their locally built binary's `-h` flag.